### ISSUE TITLE
No automated adjustment of text formatting when zooming in email window

### ISSUE THREAD

**Issue by mgw2013** [thunderbird/thunderbird-android#7678]
### Checklist

- [X] I have used the search function to see if someone else has already submitted the same bug report.
- [X] I will describe the problem with as much detail as possible.

### App version

6.603 (36003) 

### Where did you get the app from?

F-Droid

### Android version

LineageOS derivative latest Android 14 with Feb security patches

### Device model

mido

### Steps to reproduce

unified (or other) inbox
select email item
zoom into the text in the email working

### Expected behavior

automated (re)format of text displayed in email window

### Actual behavior

text (letters et al.) becomes larger but "drops" over screen edges - no adjustment of formatting - i. e. text is enlarged but layout format doesn't change 

Screenshot:

![Screenshot_20240227-175925_K-9 Mail](https://github.com/thunderbird/thunderbird-android/assets/5862935/e114664b-3c7a-4646-a699-75c145bd3496)

**Comment by ccpp** [comment]
I need this, too. I'm tired of horizontal scrolling.

Possibly related: [this open question on stack overflow](https://stackoverflow.com/questions/43720877/line-rewrapping-on-zoom)

**Comment by erdnuesse** [comment]
Just wanted to open the same issue.
Point of a mobile app for emails is actual reformatting for email readability on mobile devices. 
So in my book this app fails to deliver the #1 priority purpose of itself. There's many things I like, but I don't love having to use two apps on my phone.

**Comment by kewisch** [comment]
I believe this is the same issue as #3172, let me know if I misunderstood.

**Comment by erdnuesse** [comment]
I believe it's not a duplicate.
What is mentioned here is also non referring to long strings.

my understanding is, that rather than zooming into a prerendered HTML canvas, we'd prefer breaking every line at the end of line but by increasing the size of the text instead, but keeping the boundaries of the HTML elements well within the borders of the visible window.
(the same way FairEmail handles emails when zooming in) 

the way it is done today increases the zoom on html body at the same scale, which sets the HTML element boundaries outside the visible window.

**Comment by kewisch** [comment]
My understanding in 3172 is that we'd adjust the CSS or augment the email html to better display the email text. It talks specifically about long lines breaking the display, but also mentions a few other use cases. One thing that could fix this issue would be to force apply CSS word wrapping. I'll add a comment there.

There is a setting noted in 3172 that flips two different behaviors, but both aren't perfect. Maybe worth flipping that option to see if it improves a bit.

**Comment by mgw2013** [comment]
> I believe this is the same issue as [#3172](https://github.com/thunderbird/thunderbird-android/issues/3172), let me know if I misunderstood.

I'm afraid that the issue has nothing to do with long strings: text isn't adjusted while zooming has nothing to do with length of text string.

**Comment by kewisch** [comment]
That is a fair point, and I think both cases still fit in 3172. For normal text that is too long we need to do something to reformat it so it fits within the view. For the example above, we also need to add some CSS or webview scaling that makes things fit. I'm not trying to put this under the rug, but I think the fix for both cases is very similar in nature so it would be better to take care in that one issue.

**Comment by erdnuesse** [comment]
Triaging usually is a pm/dev task, so totally fine with me. 
Generally speaking, if an issue is not explicitly a duplicate, users are encouraged to open a new issue rather than hijacking existent ones, or at least it was common knowledge to do so in the past. 
However, for me it was just important to highlight the two separate points and "wishes" from the userbase. 

If they would be tackled during the same fix/change/story points, I'm onboard with however you see fit. As long as neither gets dropped due to perceived redundancy (the other zooming issue has been opened like half a dozen times, while this is the only one with that specific description AFAICT)

Hence I'd like to see this rather linked and open, than closed as duplicate/closed, as both issues may be fixed with one commit, because they test very very different, but in the end it wouldn't matter what I think anyway. 

Lastly; for me it's not really intuitive to follow an issue that's - to me at least - clearly different from *this*, until I can use the app properly and productively, as you could say, there exists a non-zero chance that one commit wouldn't fix the other issue. So I'd blindly trust that the other one would fix this as well. 

But let's give it a shot. Thanks for your efforts. *thumbs up*

**Comment by mgw2013** [comment]
> That is a fair point, and I think both cases still fit in 3172. For normal text that is too long we need to do something to reformat it so it fits within the view. For the example above, we also need to add some CSS or webview scaling that makes things fit. I'm not trying to put this under the rug, but I think the fix for both cases is very similar in nature so it would be better to take care in that one issue.

That is also fine with me. Let me know if and how I can help. -mgw

**Comment by kewisch** [comment]
If the fix for that issue does not fix this one we can always re-open, and I might end up being wrong. But that is ok, filing a follow-up issue or re-opening this one is cheap. As I commented on the other issue about this one I am hoping someone will read it while fixing and take care :)

If you want to debug this specific issue and have some proficiency with Kotlin code, you could check the files I linked in the other issue and see if you can adjust the webview accordingly. There is for example a TEXT_AUTOSIZING layout algorithm on API level 19 and above which could help, there are some settings regarding zoom support and overview mode. It might be necessary to adjust the <meta> tag in the HTML content if one exists, and if not then set one.

**Comment by erdnuesse** [comment]
The way I see it now, as #3172 is now completed, the behavior remains. Zooming results not in text size increase, but zooming like it was an image. Reopen?

**Comment by mgw2013** [comment]
> The way I see it now, as [#3172](https://github.com/thunderbird/thunderbird-android/issues/3172) is now completed, the behavior remains. Zooming results not in text size increase, but zooming like it was an image. Reopen?

Plus 1 - the issue is still there as best as I can tell.

**Comment by mgw2013** [comment]
@kewisch 
Here is the latest screen shot from today with the latest stable version on Android. By the way: I switched devices in between...actual shot from peridot with HyperOS2, latest security updates, no rooting and alike.

![Image](https://github.com/user-attachments/assets/4b961d18-efe8-4e23-979f-6a78a977192f)

**Comment by ccpp** [comment]
Yes, please reopen! This issue was not affected by #3172 at all.
